### DISCUSSION TITLE
[DONE] italian language

### DISCUSSION THREAD

**Discussion by AsmaR2024** [KL-Psychological-Methodology/ESMira#10] [Ideas]
please add Italian language

**Comment by SelinaDev** [comment]
Hi! Thank you for your interest! Unfortunately no one in our team speaks Italian, and we currently do not have access to Italian native speakers to translate for us.
Assuming that you speak Italian, we would be more than happy if you were to decide to contribute translations. The easiest way to help out is via our weblate instance: https://translate.jodli.dev/projects/esmira/

**Comment by JodliDev** [comment]
If you plan on translating yourself, we are happy to add italian language to the available options in our weblate platform.
If you want, we can also add machine translated suggestions to speed up your process.

**Comment by AsmaR2024** [comment]
Hi,

Thank you for the information.

We have the instructions and questionnaires in Italian language, but to run the app in Italian, translation can be done through machine translations.

Regards,
Asma

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by JodliDev** [reply]
The problem is, that translations of ESMira itself would affect all ESMira users. So we need to make sure that these translations have a good enough quality.
That means we cannot just add machine translations without checking them first.

If you need ESMira in italian language, you will have to help us translate it, or hope that we get an oportunity to do it ourselfs at some point :)

**Comment by AsmaR2024** [comment]
There is no problem. We can help in translation.

Regards,
Asma

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by JodliDev** [reply]
Great!
I added italian as a language.

First you will have to create an account at our translation platform:
https://translate.jodli.dev/

The projects that need to be translated are:
https://translate.jodli.dev/projects/esmira/apps/it/
https://translate.jodli.dev/projects/esmira/web/it/
https://translate.jodli.dev/projects/esmira/ios/it/
https://translate.jodli.dev/projects/esmira/ios_plurals/it/

If you have any questions (either regarding the process or some specific phrases) please ask away. There are probably some phrases that need further explaination or context.

I am still working on the machine translated suggestions. So it might take a couple of days until they are added :)

**Reply by JodliDev** [reply]
Ok I have added machine translated suggestions to all italian translations. You will find the translations at the bottom of each translation. To apply them press the green checkmark (or press the green pen and change it accordingly).

Again, if you need any help please let us know! :)

**Comment by AsmaR2024** [comment]
Hi,

Thank you for your support. i was trying to get registered on the translation platform but i don't receive the confirmation link and I could not sign into it.

Attached is the screenshot of the message i got when I registered.

Regards,
Asma
________________________________

**Reply by JodliDev** [reply]
I apologize. There was an error with our email server, so no emails were being sent out. Please try again. It should work now

**Comment by AsmaR2024** [comment]
Hi,

I requested to add the Italian language to the ap for the research purpose. I am unable to get back into my account and my professor has not been able to register. the problem is he can not edit the machine translation. As he's Italian he can approve the translation. Can you tell me how can he access to edit the machine translation?

Regards,
Asma

________________________________

**Reply by JodliDev** [reply]
Hello,
Can you describe what happened after confirming your email? Or do you have any more details on what is going wrong?

I can see both of your accounts and that they have been confirmed via email. But neither of them have set a password (which you should be able to, when opening the link from your confirmation email). Which is why you cannot log-in.
Are you sure that you have set a password after clicking the registration link in your email?

**Comment by AsmaR2024** [comment]
Hi,

Thank you for responding. yes, I tried to reset the password by logging in. But when I entered the password, it said wrong password or id.

Regards,
Asma
________________________________

**Reply by JodliDev** [reply]
I think you are filling out the wrong form. It looks to me that you are trying to fill out the login form instead.

When resetting your password you need to click the link in the email. This opens a webpage with the following form
![Screenshot 2024-03-05 at 10-49-54 Password reset](https://github.com/KL-Psychological-Methodology/ESMira/assets/19917795/7ef579a6-12ca-41e9-ae71-a4cee57176d4)
There you have to provide a password. Then click on "Set my password", and make sure it confirms that your password was changed (and that there are no error messages).

Then you can login.

**Comment by AsmaR2024** [comment]
Hi,

when i click on the link, it leads to me this page and i found no option to select the password change option.

Regards,
Asma
________________________________

**Comment by AsmaR2024** [comment]
Attached is the screenshot
________________________________

**Reply by JodliDev** [reply]
Unfortunately images do not seem to be displayed if you respond via email. You have to use the Github-platform instead

**Comment by AsmaR2024** [comment]
Moreover it says Try registering again to verify your identity, the verification token probably expired.
________________________________

**Reply by JodliDev** [reply]
Yes. These links expire after 48 hours. You have to use the password forgotten option again

**Reply by AsmaR2024** [reply]
I actually tried again today with the new link. and the same window appears.

![Screenshot 2024-03-05 120336](https://github.com/KL-Psychological-Methodology/ESMira/assets/159907529/5cc1d487-88eb-4eee-bea1-aace71fad9dd)

**Reply by JodliDev** [reply]
Hm. Can you try again on a different device (maybe even from a different network)? You will have to send yourself a new Forgot-password link.

I think you are using some kind of anti virus software that calls the url ahead of time (or maybe its your email software or router doing it). This means the URL is called twice. But since it can only be used once (and is invalidated after usage) you are getting this error when you actually see the web page.
This should not happen and is bad practice. But I have seen anti virus software doing it.

**Comment by AsmaR2024** [comment]
Hi,

 Can you please assist with how much space the server needs for ESMira?
 
Like it says that  we need to create our own server here:  https://esmira.kl.ac.at/?lang=en#home

Regards,
Asma

**Reply by JodliDev** [reply]
See https://github.com/KL-Psychological-Methodology/ESMira/discussions/13

**Comment by JodliDev** [comment]
I see translating to italian seems to go well. Thank you very much! :)
Did you encounter any issues with translations?

We are in the process of adding acknowledgementes for translations on the starting page of the GitHub project. Would that be OK with you?
Wo do you want us to cite? (it does not have to be only one person)

**Comment by AsmaR2024** [comment]
Hi,

 I did not add a translation of the app as Italian is not my native language. It was my supervisor Sebastiano Costa whom I requested to translate.

Regards,
Asma
________________________________

**Reply by JodliDev** [reply]
Would it be ok for Sebastiano Costa to be in the acknowledgements?

**Comment by AsmaR2024** [comment]
Hi,

I asked my professor, he said you can add his name ( Sebastiano Costa) along with mine in the acknowledgment. 🙂

Regards,
Asma
________________________________

**Comment by SelinaDev** [comment]
Just an update on the Language situation: We uploaded new versions of both the Android and iOS Apps today, both should include the new Italian translation (Android App is already available in the Play Store, iOS might take a bit longer, but should be available soon as well). We plan to release an update for ESMira-web either later this week or next week, which will include the translation for the online-platform.
